### DISCUSSION TITLE
Predicting to dataset

### DISCUSSION THREAD

**Discussion by franciskasara** [opengeos/segment-geospatial#75] [Q&A] [answered]
What would be the most ideal / computationally cheap way to predict to several images? A simple for cycle or can the "image" parameter in sam.generate somehow be a datagenerator?

**Comment by giswqs** [comment, marked as answer]
For now, you can use a loop to segment multiple images. The generate function might be able to accept multiple images in the future. See https://github.com/facebookresearch/segment-anything/pull/390

**Reply by adm-gis** [reply]
Hi, thank you for the response.  I have a similar question below.

I have a large satellite image that overwhelms our GPU (10gb).  I was able to tile the image and then loop over the tiles with the generate function.  However, since I am only interested in segmenting the building footprints, I'd like to use the predict function, using an input layer.  Is there a way to use the input from one tile and apply that input to the subsequent tiles?  

Thanks!
